CMakeLists.txt

```
@@ -1,19 +1,51 @@
 cmake_minimum_required(VERSION 3.1)
 
-project(dbg_macro)
+project(dbg_macro LANGUAGES CXX)
 
-set(CMAKE_CXX_STANDARD 11)
+###############################################################################
+# CMake Options
+###############################################################################
+
+option(DBG_MACRO_BUILD_EXAMPLES OFF)
+option(DBG_MACRO_BUILD_TESTS OFF)
+
+###############################################################################
+# Add dbg_macro Library
+###############################################################################
 
 add_library(${PROJECT_NAME} INTERFACE)
-target_include_directories(${PROJECT_NAME} INTERFACE ".")
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
+target_include_directories(${PROJECT_NAME} INTERFACE
+	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+	$<INSTALL_INTERFACE:include>)
+
```

```suggestion
target_compile_features(${PROJECT_NAME} INTERFACE cxx_std_11)
```
Make the compile feature part of the target's interface. You should then not need to do this in the example and tests.

`cxx_std_11` is also a 3.8 feature

https://cmake.org/cmake/help/v3.7/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html

---

Yeah, you are right! I'll do it right now.

---

Done! At first I had some trouble dealing with an old cmake version on Travis CI, that's why I was still using cmake 3.1 even though it was completely wrong.

---

Sorry, I didn't know about Commit Suggestion feature on GitHub. I commited your recommendation though as you requested.

---

Neither did I until GitHub prompted it :P

---

Looks super useful! :100: